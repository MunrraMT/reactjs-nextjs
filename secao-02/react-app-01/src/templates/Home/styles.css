.container {
  min-height: calc(100vh - 6rem);
  background-color: #eee;
  padding: 1rem;
}
