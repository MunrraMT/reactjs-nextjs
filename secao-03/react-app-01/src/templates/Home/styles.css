.container {
  min-height: calc(100vh - 10rem);
  background-color: #eee;
  padding: 1rem;
}
