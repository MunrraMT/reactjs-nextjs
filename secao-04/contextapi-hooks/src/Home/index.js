import { useState } from 'react';
import Button from '../components/Button';
import Header from '../components/Header';
import { useCounterContext } from '../contexts/CounterContext';

const Home = () => {
  const [state, actions] = useCounterContext();

  const [value, setValue] = useState('');

  return (
    <main>
      <Header />

      <Button onBtnClick={actions.increase}>Increase</Button>
      <Button onBtnClick={actions.decrease}>Decrease</Button>
      <Button onBtnClick={actions.reset}>Reset</Button>
    </main>
  );
};

export default Home;
