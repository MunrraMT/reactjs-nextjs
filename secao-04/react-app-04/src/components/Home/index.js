import { lazy, Suspense, useState } from 'react';

const loadComponent = () => {
  console.log('Carregando componente');

  return import('./lazy-component');
};

const LazyComponent = lazy(loadComponent);

const NewHome = () => {
  const [show, setShow] = useState(false);

  return (
    <section>
      <button
        type="button"
        onMouseOver={loadComponent}
        onClick={() => setShow((prev) => !prev)}
      >
        {show ? 'LC is on screen' : 'LC is off screen'}
      </button>
      <Suspense fallback={<p>Loading...</p>}>
        {show && <LazyComponent />}
      </Suspense>
    </section>
  );
};

export default NewHome;
