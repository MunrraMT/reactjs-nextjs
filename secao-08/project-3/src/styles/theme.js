const theme = {
  colors: {
    mainBg: 'red',
    secondaryBG: 'green',
  },
};

export default theme;
