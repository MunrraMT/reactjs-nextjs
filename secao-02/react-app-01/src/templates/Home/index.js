import './styles.css';

import { useEffect, useState } from 'react';

import loadPosts from '../../utils/load-posts';
import Posts from '../../components/Posts';
import Button from '../../components/Button';
import TextInput from '../../components/TextInput';

const Home = () => {
  const [posts, setPosts] = useState([]);
  const [allPosts, setaAllPosts] = useState([]);
  const [page, setPage] = useState(0);
  const [postsPerPage] = useState(3);
  const [searchValue, setSearchValue] = useState('');

  useEffect(() => {
    const handleLoadPosts = async () => {
      const postsAndPhotos = await loadPosts();

      setPosts(postsAndPhotos.slice(page, postsPerPage));
      setaAllPosts(postsAndPhotos);
    };

    handleLoadPosts();
  }, []);

  const loadMorePosts = () => {
    const nextPage = page + postsPerPage;
    const nextPosts = allPosts.slice(nextPage, nextPage + postsPerPage);

    setPosts([...posts, ...nextPosts]);
    setPage(nextPage);
  };

  const handleInput = (e) => {
    const { value } = e.target;
    setSearchValue(value);
  };

  const noMorePosts = posts.length === allPosts.length;

  const filteredPosts = allPosts.filter((post) => {
    return (
      post.title.toLowerCase().indexOf(searchValue.toLowerCase()) !== -1 ||
      post.body.toLowerCase().indexOf(searchValue.toLowerCase()) !== -1
    );
  });

  return (
    <>
      <header
        style={{
          display: 'flex',
          flexDirection: 'column',
          alignItems: 'center',
          justifyContent: 'center',
          padding: '1rem',
          height: '7rem'
        }}
      >
        <h1>Estudo de ReactJS - App Posts</h1>
        <TextInput searchValue={searchValue} handleInput={handleInput} />
      </header>

      <main className='container'>
        {searchValue ? (
          (filteredPosts.length === 0 && <p>Busca não encontrada</p>) ||
          (filteredPosts.length > 0 && <Posts posts={filteredPosts} />)
        ) : (
          <Posts posts={posts} />
        )}
      </main>

      {!searchValue && (
        <footer>
          <Button disabled={noMorePosts} handleClick={loadMorePosts}>
            Ver mais posts
          </Button>
        </footer>
      )}
    </>
  );
};

// class Home2 extends Component {
//   state = {
//     posts: [],
//     allPosts: [],
//     page: 0,
//     postsPerPage: 3,
//     searchValue: ''
//   };

//   loadPosts = async () => {
//     const { page, postsPerPage } = this.state;

//     const postsAndPhotos = await loadPosts();

//     this.setState({
//       posts: postsAndPhotos.slice(page, postsPerPage),
//       allPosts: postsAndPhotos
//     });
//   };

//   loadMorePosts = () => {
//     const { page, postsPerPage, allPosts, posts } = this.state;

//     const nextPage = page + postsPerPage;
//     const nextPosts = allPosts.slice(nextPage, nextPage + postsPerPage);

//     posts.push(...nextPosts);

//     this.setState({ posts, page: nextPage });
//   };

//   handleInput = (e) => {
//     const { value } = e.target;
//     this.setState({ searchValue: value });
//   };

//   async componentDidMount() {
//     await this.loadPosts();
//   }

//   componentDidUpdate() {
//     console.log(this.props['test-props']);
//   }

//   render() {
//     const { posts, allPosts, searchValue } = this.state;

//     const noMorePosts = posts.length === allPosts.length;

//     const filteredPosts = allPosts.filter((post) => {
//       return (
//         post.title.toLowerCase().indexOf(searchValue.toLowerCase()) !== -1 ||
//         post.body.toLowerCase().indexOf(searchValue.toLowerCase()) !== -1
//       );
//     });

//     return (
//       <>
//         <header
//           style={{
//             display: 'flex',
//             flexDirection: 'column',
//             alignItems: 'center',
//             justifyContent: 'center',
//             padding: '1rem',
//             height: '7rem'
//           }}
//         >
//           <h1>Estudo de ReactJS - App Posts</h1>
//           <TextInput searchValue={searchValue} handleInput={this.handleInput} />
//         </header>

//         <main className='container'>
//           {searchValue ? (
//             (filteredPosts.length === 0 && <p>Busca não encontrada</p>) ||
//             (filteredPosts.length > 0 && <Posts posts={filteredPosts} />)
//           ) : (
//             <Posts posts={posts} />
//           )}
//         </main>

//         {!searchValue && (
//           <footer>
//             <Button disabled={noMorePosts} handleClick={this.loadMorePosts}>
//               Ver mais posts
//             </Button>
//           </footer>
//         )}
//       </>
//     );
//   }
// }

// function App() {
//   return (
//     <div className="App">
//       <header className="App-header">
//         <img src={logo} className="App-logo" alt="logo" />
//         <p>
//           Edit <code>src/App.js</code> and save to reload.
//         </p>
//         <a
//           className="App-link"
//           href="https://reactjs.org"
//           target="_blank"
//           rel="noopener noreferrer"
//         >
//           Learn React
//         </a>
//       </header>
//     </div>
//   );
// }

export default Home;
