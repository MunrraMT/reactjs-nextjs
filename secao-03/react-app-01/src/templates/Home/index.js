import './styles.css';

import { useEffect, useState } from 'react';

import loadPosts from '../../utils/load-posts';
import Posts from '../../components/Posts';
import Button from '../../components/Button';
import TextInput from '../../components/TextInput';

const Home = () => {
  const [posts, setPosts] = useState([]);
  const [allPosts, setaAllPosts] = useState([]);
  const [page, setPage] = useState(0);
  const [postsPerPage] = useState(3);
  const [searchValue, setSearchValue] = useState('');

  useEffect(() => {
    let isCancelled = false;

    loadPosts().then((postsAndPhotos) => {
      if (!isCancelled) {
        setPosts(postsAndPhotos.slice(page, postsPerPage));
        setaAllPosts(postsAndPhotos);
      }
    });

    return () => {
      isCancelled = true;
    };
  }, []);

  const loadMorePosts = () => {
    const nextPage = page + postsPerPage;
    const nextPosts = allPosts.slice(nextPage, nextPage + postsPerPage);

    setPosts((prevPosts) => [...prevPosts, ...nextPosts]);
    setPage(nextPage);
  };

  const handleInput = (e) => {
    const { value } = e.target;
    setSearchValue(value);
  };

  const noMorePosts = posts.length === allPosts.length;

  const filteredPosts = allPosts.filter(
    (post) =>
      post.title.toLowerCase().indexOf(searchValue.toLowerCase()) !== -1,
  );

  return (
    <>
      <header
        style={{
          display: 'flex',
          flexDirection: 'column',
          alignItems: 'center',
          justifyContent: 'center',
          padding: '1rem',
          height: '7rem',
        }}
      >
        <h1>Estudo de ReactJS - App Posts</h1>
        <TextInput searchValue={searchValue} handleInput={handleInput} />
      </header>

      <main className='container'>
        {searchValue ? (
          (filteredPosts.length === 0 && <p>Busca n√£o encontrada</p>) ||
          (filteredPosts.length > 0 && <Posts posts={filteredPosts} />)
        ) : (
          <Posts posts={posts} />
        )}
      </main>

      {!searchValue && (
        <footer>
          <Button
            disabled={noMorePosts}
            handleClick={loadMorePosts}
            text='Ver mais posts'
          />
        </footer>
      )}
    </>
  );
};

export default Home;
