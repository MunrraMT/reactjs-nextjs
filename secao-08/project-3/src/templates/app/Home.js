import * as Styled from './styles';

function Home() {
  return (
    <Styled.Wrapper>
      <h1>Hello</h1>
    </Styled.Wrapper>
  );
}

export default Home;
