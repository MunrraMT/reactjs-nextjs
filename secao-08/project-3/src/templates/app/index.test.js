import { screen } from '@testing-library/react';

import renderTheme from '../../styles/render-theme';
import Home from '.';
import theme from '../../styles/theme';

describe('Home', () => {
  test('testando comunicação com tema', () => {
    renderTheme(<Home />);
    const headingContainer = screen.getByTestId('wrapper');
    expect(headingContainer).toHaveStyle({
      background: theme.colors.mainBg,
    });
    expect(headingContainer).toMatchSnapshot();
  });
});
