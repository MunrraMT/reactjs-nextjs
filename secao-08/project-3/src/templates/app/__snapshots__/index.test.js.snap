// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home testando comunicação com tema 1`] = `
.c0 {
  background: red;
}

<header
  class="c0"
  data-testid="wrapper"
>
  <h1>
    Hello
  </h1>
</header>
`;
